src/core/device.h

```diff
@@ -11,26 +11,29 @@
 
 typedef struct Device Device;
 
+/* A mask specifying where we have seen the device currently. This is a bitmask because the device might show up
+ * asynchronously from each other at various places. For example, in very common case a device might already be mounted
+ * before udev finished probing it (think: a script setting up a loopback block device, formatting it and mounting it
+ * in quick succession). Hence we need to track precisely where it is already visible and where not. */
 typedef enum DeviceFound {
-        DEVICE_NOT_FOUND     = 0,
-        DEVICE_FOUND_UDEV    = 1 << 1,
-        DEVICE_FOUND_UDEV_DB = 1 << 2,
-        DEVICE_FOUND_MOUNT   = 1 << 3,
-        DEVICE_FOUND_SWAP    = 1 << 4,
+        DEVICE_NOT_FOUND   = 0,
+        DEVICE_FOUND_UDEV  = 1U << 1, /* The device has shown up in the udev database */
+        DEVICE_FOUND_MOUNT = 1U << 2, /* The device has shown up in /proc/self/mountinfo */
+        DEVICE_FOUND_SWAP  = 1U << 3, /* The device has shown up in /proc/swaps */
```

Comment:
The type of rvalue here has no effect on the enum type (unless the value is large enough to force a certain width, but that's not the case here), so "U" is just misleading.

Reply:
so, you are right, this won't have any effect on sizeof(DEVICE_FOUND_UDEV). However, it does have an effect on the shift-left sooner or later. "1 << 31" is undefined in C on 32bit systems, while "1U << 31" is well defined. Now, you might say that "that's a synthetic case, and we never have so many flags defined" and as soon as you shift left by 32 you need "1ULL << 32" anyway, but I think in general, signed-extended shift-left is really really weird, and I'd rather not do that if we are talking about a flags sets...

Reply:
I think that's too much magic. Not in a case like this, but we often define masks and "_MAX" and such, and it'd be awkward to combine with a macro. So I think it's fine as is.

Reply:
Hmm, by the time I read this i already had done #9224... :-(

PTAL

(I'd be fine if this isn't merged, but I kinda like the unifying work this does, after all we currently use 1 <<, 1U <<, 1UL <<, 1ULL <<, and UINT64_C(1) << at different places all over the place...